wandb/sdk/internal/internal.py

```diff
@@ -36,10 +36,27 @@
 from ..interface import interface
 
 
+if wandb.TYPE_CHECKING:
```

Comment:
I am sure you are away of this, but `wandb.TYPE_CHECKING` is a bit of a mis-nomer. Under the hood it really is "python verison is >= 3.6", and logically means "retains_type_annotations after stripping process". While the common `TYPE_CHECKING` [variable is](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) "A special constant that is assumed to be True by 3rd party static type checkers. It is False at runtime". Anyway, just thought i would point out this potential naming confusion here, although it is used in many places.

Reply:
Yes, it is confusing, but it is used to guard the import typing that follows most cases.

The correct correct way to use this right now is:
```
if wandb.TYPE_CHECKING
   from typing import TYPE_CHECKING
   if TYPE_CHECKING:
         # do stuff that only runs during type checking
   # add other stuff here to run in py3.6+
```

You should try to put all your type checking under `if TYPE_CHECKING`
but then you have to use string types, not actual objects. (unless we are using py3.8 and beyond and using from future annotations)

I will write up a guide so at least we can be consistent.   I guess I can change wandb.TYPE_CHECKING to be something less likely to be confused.

Reply:
Cool - thanks for the details here. I'll make sure to do the double guard for datatypes and others.